Adding more functions all, any, get_dtype, and pass through functions.

ankur-gupta opened the pull request.

yulyt commented on flicker/flicker.py:
```diff
@@ -204,6 +204,21 @@ def _reset(self, df):
+            names = self._df.columns
+        if name not in names:
+            if dataframe_name is None:
+                msg = 'column name "{}" not found'
```
I think you're missing a `msg.format(name)` for this?

ankur-gupta replied:
You're right. Thanks!

yulyt commented on flicker/flicker.py:
```diff
@@ -204,6 +204,21 @@ def _reset(self, df):
         self._nrows = None
         self._ncols = None
 
+    def _validate_column_name(self, name, names=None, dataframe_name=None):
```
Add docstring?

ankur-gupta replied:
Added!

yulyt commented on flicker/flicker.py:
```diff
@@ -204,6 +204,21 @@ def _reset(self, df):
+            raise TypeError(msg)
+        if names is None:
+            names = self._df.columns
+        if name not in names:
```
If `names` is None, this will raise an exception. Was an `elif` meant to be used here?

yulyt commented on flicker/flicker.py:
```diff
@@ -985,7 +1000,11 @@ def value_counts(self, names, normalize=False, sort=True, ascending=False,
         if isinstance(names, six.string_types):
             names = [names]
         for name in names:
```
Could you use the `_validate_column_name` method here too?

ankur-gupta replied:
Thanks!

yulyt commented on flicker/flicker.py:
```diff
@@ -1409,16 +1463,16 @@ def join(self, other, on=None, how='inner',
-                msg = ('column name "{}" not found in the {} dataframe')
-                msg = msg.format(name, dataframe)
-                raise KeyError(msg)
+        # def _validate_column_name(name, names, dataframe):
```
rm?

ankur-gupta replied:
Fixed, thanks :)

ankur-gupta commented:
@yulyt Thanks for you review! I have addressed all of your comments. I've also added tests to increase the coverage from ~52% to ~61%. More tests need to be written (especially for `value_counts()` which has a known issue) but we're making progress. 

If you can, please take another look. If not, it's okay. I'll merge it later today.

Once we merge this PR, I would like to add a `CONTRIBUTING.md` file and replace my name with `Flicker Contributors` in all copyright notices. We already have an Apache 2.0 license (same as spark).
